### PR TITLE
using getBetterSlot for different prebuffed skills

### PR DESCRIPTION
**Pull request by mf2212**
it is analogous with enchant, checking the higher level of the Energy Shield skill on both weapon tab, before precasting it.

### CODE REVIEW

File: d2bs/kolbot/libs/common/Precast.js [hunk 1]
```diff
@@ -474,4 +475,23 @@ MainLoop:
 
 		return true;
 	};
+
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Precast.js]: This function is not needed, Precast already includes function: precastSkill which does the same thing as your energyshield function does.

File: d2bs/kolbot/libs/common/Precast.js [hunk 2]
```diff
@@ -159,7 +160,7 @@ var Precast = new function () {
 			}
 
 			if (!me.getState(30) || force) {
-				Skill.cast(58, 0); // Energy Shield
+				this.energyshield(); //Energy Shield
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Precast.js]: Replace with precastSkill

File: d2bs/kolbot/libs/common/Attacks/Sorceress.js [hunk 3]
```diff
@@ -12,7 +12,7 @@ var ClassAttack = {
 		}
 
 		if (!me.getState(30) && me.getSkill(58, 1)) {
-			Skill.cast(58, 0);
+            Precast.energyshield(); //Energy Shield
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Attacks/Sorceress.js]: Replace with precastSkill, fix tab formatting and comment formatting.

**Comment by mf2212** [comment]
ok
Attacks/Sorceress.js
should be `            Precast.precastSkill(58); //Energy Shield`
and in Precast.js the energyshield function is useless, but there it should be the precastskill(58) too
`			if (!me.getState(30) || force) {
				this.precastSkill(58); //Energy Shield
			}`

File: d2bs/kolbot/libs/common/Attacks/Sorceress.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ var ClassAttack = {
 		}
 
 		if (!me.getState(30) && me.getSkill(58, 1)) {
-			Skill.cast(58, 0);
+            Precast.precastSkill(58); //Energy Shield
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Attacks/Sorceress.js]: Fix tabs preceding Precast, add space between // and Energy Shield

File: d2bs/kolbot/libs/common/Precast.js [hunk 5]
```diff
@@ -159,7 +154,7 @@ var Precast = new function () {
 			}
 
 			if (!me.getState(30) || force) {
-				Skill.cast(58, 0); // Energy Shield
+				this.precastSkill(58); //Energy Shield
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Precast.js]: Add space between // and Energy Shield

**Comment by noah-** [comment]
I posted to discord for testing. One concern I have is in the Sorc script, maybe it should remain Skill.cast instead of tab check because that recasts ES in the middle of attacks and swapping tab might cause lower stats/death. Will need testing to confirm.

**Comment by mf2212** [comment]
btw these can be extended for Thunder Storm and other Sorceress prebuffed skills

**Comment by noah-** [comment]
Feel free to extend to TS and any other prebuffed skills you like. Also, I think for now remove the check in sorceress.js unless you add some checks to see sorc is in a safe place to do a weapon swap (ie no monsters nearby or something).

**Comment by mf2212** [comment]
I added:
- sorceress: cold armors and TS
- necromancer BA
- druid CA
- assassin BS, BoS and Fade

**Comment by jaenster** [comment]
I tested this, it works great! =)

**Pull request merged**